client/rest-high-level/src/main/java/org/elasticsearch/client/indices/SimulateIndexTemplateRequest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.indices;
+
+import org.elasticsearch.client.TimedRequest;
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+
+import java.io.IOException;
+
+/**
+ * A request to simulate matching a provided index name and an optional new index template against the existing index templates.
+ */
+public class SimulateIndexTemplateRequest extends TimedRequest implements ToXContentObject {
+
+    private static final ParseField INDEX_NAME = new ParseField("index_name");
+    private static final ParseField INDEX_TEMPLATE = new ParseField("index_template");
+
+    private String indexName;
+
+    @Nullable
+    private PutIndexTemplateV2Request indexTemplateV2Request;
+
+    public SimulateIndexTemplateRequest(String indexName) {
+        if (Strings.isNullOrEmpty(indexName)) {
+            throw new IllegalArgumentException("index name cannot be null or empty");
+        }
+        this.indexName = indexName;
+    }
+
+    /**
+     * Return the index name for which we simulate the index template matching.
+     */
+    public String indexName() {
+        return indexName;
+    }
+
+    /**
+     * Set the index name to simulate template matching against the index templates in the system.
+     */
+    public SimulateIndexTemplateRequest indexName(String indexName) {
+        this.indexName = indexName;
```

Comment:
We validate the `indexName` in the constructor but maybe we should also validate it here?